src/canvas/canvasRenderer.js

```diff
@@ -0,0 +1,166 @@
+//////////////////////////////////////////////////////////////////////////////
+/**
+ * Create a new instance of class canvasRenderer
+ *
+ * @class
+ * @extends geo.renderer
+ * @param canvas
+ * @returns {geo.gl.canvasRenderer}
+ */
+//////////////////////////////////////////////////////////////////////////////
+geo.canvas.canvasRenderer = function (arg) {
+  'use strict';
+
+  if (!(this instanceof geo.canvas.canvasRenderer)) {
+    return new geo.canvas.canvasRenderer(arg);
+  }
+  arg = arg || {};
+  geo.renderer.call(this, arg);
+
+  var m_this = this,
+      m_width = 0,
+      m_height = 0,
+      m_renderAnimFrameRef = null,
+      s_init = this._init,
+      s_exit = this._exit;
+
+  // ccl: is there a batter way of gettign the layer in render?
+  m_this.ccl_layer = arg.layer;
```

Comment:
I added this before I knew how to get the layer from the rendering.  It is not used, and has been removed.